src/Workspaces/Core/Portable/CodeStyle/CodeStyleHelpers.cs

```diff
@@ -1,52 +1,94 @@
                     {
-                        option = new CodeStyleOption<bool>(value: isEnabled, notification: NotificationOption.None);
+                        notificationOpt = notificationOpt ?? NotificationOption.None;
+                        option = new CodeStyleOption<bool>(false, notificationOpt);
                         return true;
                     }
-                    else
+                    else if (notificationOpt != null)
                     {
-                        option = CodeStyleOption<bool>.Default;
-                        return false;
+                        option = new CodeStyleOption<bool>(true, notificationOpt);
+                        return true;
                     }
                 }
-                option = CodeStyleOption<bool>.Default;
-                return false;
             }
 
-            if (!bool.TryParse(args[0].Trim(), out isEnabled))
+            option = CodeStyleOption<bool>.Default;
+            return false;
+        }
+
+        /// <summary>
+        /// Given an editor-config code-style-option, gives back the constituent parts of the 
+        /// option.  For example, if the option is "true:error" then "true" will be returned
+        /// in <paramref name="value"/> and <see cref="NotificationOption.Error"/> will be returned
+        /// in <paramref name="notificationOpt"/>.  Note that users are allowed to not provide
+        /// a NotificationOption, so <paramref name="notificationOpt"/> may be null.  The caller
+        /// of this method must decide what to do in that case.
+        /// </summary>
+        public static bool TryGetCodeStyleValueAndOptionalNotification(
+            string arg, out string value, out NotificationOption notificationOpt)
+        {
+            var args = arg.Split(':');
+            Debug.Assert(args.Length > 0);
+
+            // We allow a single value to be provided in some cases.  For example users 
+            // can provide 'false' without supplying a notification as well.  Allow the 
+            // caller of this to determine what to do in this case.
+            if (args.Length == 1)
+            {
+                value = args[0];
+                notificationOpt = null;
+                return true;
+            }
+
+            if (args.Length == 2)
             {
-                option = CodeStyleOption<bool>.Default;
-                return false;
+                // If we have two args, then the second must be a notification option.  If 
+                // it isn't, then this isn't a valid code style option at all.
+                if (TryParseNotification(args[1], out var localNotification))
+                {
+                    value = args[0];
+                    notificationOpt = localNotification;
+                    return true;
+                }
             }
 
-            switch (args[1].Trim())
+            // We only support 0 or 1 args.  Anything else can't be parsed properly.
```

Comment:
@jasonmalinowski Something to think about for our editorconfig work.  If we could just have the editorconfig be an 'additional file' (for example), we could write an analyzer that gave warning for values that we don't support for our keys.